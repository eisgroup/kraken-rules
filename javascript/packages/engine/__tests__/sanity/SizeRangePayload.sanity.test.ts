/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { sanityEngine } from "./_SanityEngine";
import { sanityMocks } from "./_AutoPolicyObject.mocks";

describe("SizeRange payload sanity test", () => {
    it("should fail when size is less than min", () => {
        const policy = sanityMocks.empty();
        policy.policies = ["1"];
        const result = sanityEngine.evaluate(policy, "SizeRangePayload");
        expect(result).k_toMatchResultsStats({ total: 1, critical: 1 });
        expect(result).k_toHaveExpressionsFailures(0);
        expect(result).k_toMatchResultsSnapshots();
    });
    it("should fail when size is more than max", () => {
        const policy = sanityMocks.empty();
        policy.policies = ["1", "2", "3", "4"];
        const result = sanityEngine.evaluate(policy, "SizeRangePayload");
        expect(result).k_toMatchResultsStats({ total: 1, critical: 1 });
        expect(result).k_toHaveExpressionsFailures(0);
        expect(result).k_toMatchResultsSnapshots();
    });
    it("should succeed when size is in range", () => {
        const policy = sanityMocks.empty();
        policy.policies = ["1", "2"];
        const result = sanityEngine.evaluate(policy, "SizeRangePayload");
        expect(result).k_toMatchResultsStats({ total: 1, critical: 0 });
        expect(result).k_toHaveExpressionsFailures(0);
        expect(result).k_toMatchResultsSnapshots();
    });
});
