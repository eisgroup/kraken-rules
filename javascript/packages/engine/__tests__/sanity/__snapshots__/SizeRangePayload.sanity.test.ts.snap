// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeRange payload sanity test should fail when size is less than min 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Policy:0:policies": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "0",
          "contextName": "Policy",
          "fieldName": "policies",
          "fieldPath": "policies",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "max": 3,
              "message": undefined,
              "min": 2,
              "success": false,
              "type": 4,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Policy",
              "payloadtype": "SIZE_RANGE",
              "ruleName": "SizePayloadRange",
              "targetPath": "policies",
            },
          },
        ],
      },
    },
  },
  "fieldResults": Object {
    "Policy:0:policies": Object {
      "fieldType": "STRING",
      "id": "Policy:0:policies",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "policies",
      "ruleResults": Array [
        Object {
          "errorCode": "rule-size-range-error",
          "errorMessage": "Array length is invalid",
          "isFailed": true,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "max": 3,
            "message": undefined,
            "min": 2,
            "success": false,
            "type": 4,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "SizePayloadRange",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`SizeRange payload sanity test should fail when size is more than max 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Policy:0:policies": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "0",
          "contextName": "Policy",
          "fieldName": "policies",
          "fieldPath": "policies",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "max": 3,
              "message": undefined,
              "min": 2,
              "success": false,
              "type": 4,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Policy",
              "payloadtype": "SIZE_RANGE",
              "ruleName": "SizePayloadRange",
              "targetPath": "policies",
            },
          },
        ],
      },
    },
  },
  "fieldResults": Object {
    "Policy:0:policies": Object {
      "fieldType": "STRING",
      "id": "Policy:0:policies",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "policies",
      "ruleResults": Array [
        Object {
          "errorCode": "rule-size-range-error",
          "errorMessage": "Array length is invalid",
          "isFailed": true,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "max": 3,
            "message": undefined,
            "min": 2,
            "success": false,
            "type": 4,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "SizePayloadRange",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`SizeRange payload sanity test should succeed when size is in range 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Policy:0:policies": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "0",
          "contextName": "Policy",
          "fieldName": "policies",
          "fieldPath": "policies",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "max": 3,
              "message": undefined,
              "min": 2,
              "success": true,
              "type": 4,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Policy",
              "payloadtype": "SIZE_RANGE",
              "ruleName": "SizePayloadRange",
              "targetPath": "policies",
            },
          },
        ],
      },
    },
  },
  "fieldResults": Object {
    "Policy:0:policies": Object {
      "fieldType": "STRING",
      "id": "Policy:0:policies",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "policies",
      "ruleResults": Array [
        Object {
          "errorCode": "rule-size-range-error",
          "errorMessage": "Array length is invalid",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "max": 3,
            "message": undefined,
            "min": 2,
            "success": true,
            "type": 4,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "SizePayloadRange",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;
